pytorch_lightning/core/lightning.py

```diff
@@ -990,6 +990,30 @@ def predict(self, batch: Any, batch_idx: int, dataloader_idx: Optional[int] = No
         """
         return self(batch)
 
+    def configure_callbacks(self):
+        """
+        Configure model-specific callbacks.
+        When the model gets attached, e.g., when ``.fit()`` or ``.test()`` gets called,
```

Comment:
still don't know the use-case for `configure_callbacks` yet, but will it be good to have an argument similar to `stage` in the `setup` with `'fit'` and `'test'`?

Reply:
I like that idea!

The use-case is supporting out-of-the-box LightningModules which require specific callbacks to work/train properly

Reply:
it makes sense. Would any of you like to add it as a follow up later? 

I'm hesitating to do it in this PR, because currently we have a bit of a confusing definition of "stages". There is now a "running_stage" and then there is the regular "stage" fit and test, and recently a _predict stage was introduced. 

Any thoughts?

Reply:
yes, in a separate one of course. We can just do it as it will be done with the setup here: https://github.com/PyTorchLightning/pytorch-lightning/issues/5658

Also, there is another way around it. We can just let users use `self.running_stage` (make it a property and return `trainer.running_stage`) and won't have any stage argument anywhere.

Although I'd still prefer having an argument, but just a thought.

Reply:
Hmm, if we don't add the stage arguments in this PR and decide to add it later, it will unfortunately be a breaking api change for the user.

Btw I tried the property approach here: #5724
However, it seems there is a bit of a mess going on hehe, it might be not so straightforward :)

Reply:
@rohitgr7 n00b question: why does the stage factor in here? The callback interface doesn't have the same pattern; it's just a flat list of events that the user can trigger on

Reply:
Ananth has a point. A callback that is only supposed to run say during test would just implement the test callback methods anyway.

Reply:
we do have some hooks that runs in both (fit and test) like `on_init_start/end`, `on_fit_start/end`.

Reply:
@rohitgr7 in that case should we add `stage` to those callback hooks directly?